controllers/storageconsumer/storageconsumer_upgrade_controller.go

```diff
@@ -257,8 +276,10 @@ func (r *StorageConsumerUpgradeReconciler) removeStorageRequestOwner(ctx context
 		if idx := slices.IndexFunc(refs, func(owner metav1.OwnerReference) bool {
 			return owner.Kind == "StorageRequest"
 		}); idx != -1 {
-			obj.SetOwnerReferences(slices.Delete(refs, idx, idx+1))
-			if err := r.Client.Patch(ctx, obj, client.MergeFrom(obj)); client.IgnoreNotFound(err) != nil {
+			objCopy := &metav1.PartialObjectMetadata{}
+			obj.DeepCopyInto(objCopy)
+			objCopy.SetOwnerReferences(slices.Delete(refs, idx, idx+1))
+			if err := r.Client.Patch(ctx, objCopy, client.MergeFrom(obj)); client.IgnoreNotFound(err) != nil {
```

Comment:
remove ignoring the error here as you are only doing the patch after ensuring the object.